src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes/Block/index.tsx

```diff
@@ -0,0 +1,320 @@
+  getFlows: () => Flow[]
+  updateFlowNode: (args: any) => void
+}
+
+const defaultLabels = {
+  action: 'studio.flow.node.chatbotExecutes',
+  execute: 'studio.flow.node.chatbotExecutes',
+  failure: 'studio.flow.node.workflowFails',
+  router: 'if',
+  listen: 'listen',
+  say_something: 'studio.flow.node.chatbotSays',
+  success: 'studio.flow.node.workflowSucceeds',
+  trigger: 'studio.flow.node.triggeredBy'
+}
+
+const BlockWidget: FC<Props> = ({
+  node,
+  editNodeItem,
+  onDeleteSelectedElements,
+  onCopySelectedElement,
+  selectedNodeItem,
+  getConditions,
+  switchFlowNode,
+  updateFlowNode,
+  getLanguage,
+  getExpandedNodes,
+  setExpanded,
+  getDebugInfo,
+  editTriggers,
+  disconnectNode
+}) => {
+  const { nodeType } = node
+  const { currentLang, defaultLang } = getLanguage()
+
+  const handleContextMenu = e => {
+    e.stopPropagation()
+    e.preventDefault()
+
+    if (defaultLang && defaultLang !== currentLang) {
+      toast.info('studio.flow.cannotAddContent')
+      return
+    }
+
+    switchFlowNode(node.id)
+    contextMenu(
+      e,
+      <Menu>
+        <MenuItem
+          icon="trash"
+          text={
+            <div className={sharedStyle.contextMenuLabel}>
+              {lang.tr('delete')}
+              <ShortcutLabel light keys={['backspace']} />
+            </div>
+          }
+          intent={Intent.DANGER}
+          onClick={onDeleteSelectedElements}
+        />
+        <MenuItem
+          icon="duplicate"
+          text={<div className={sharedStyle.contextMenuLabel}>{lang.tr('copy')}</div>}
+          onClick={() => onCopySelectedElement(node.id)}
+        />
+        <MenuDivider />
+        {nodeType === 'trigger' && <MenuItem icon="edit" text={lang.tr('edit')} onClick={() => editTriggers(node)} />}
+        <MenuItem icon="minimize" text={lang.tr('studio.flow.disconnectNode')} onClick={() => disconnectNode(node)} />
+        {nodeType === 'router' ? (
+          <React.Fragment>
+            <MenuDivider />
+            <MenuItem text={lang.tr('studio.flow.chips')}>
+              <MenuItem
+                text={lang.tr('studio.flow.transition')}
+                onClick={async () => {
+                  await switchFlowNode(node.id)
+                  updateFlowNode({ next: [...node.next, defaultTransition] })
+                }}
+                icon="flow-end"
+              />
+            </MenuItem>
+          </React.Fragment>
+        ) : null}
+      </Menu>
+    )
+  }
+
+  const inputPortInHeader = !['trigger'].includes(nodeType)
+  const outPortInHeader = !['failure', 'router', 'success'].includes(nodeType)
+  const canCollapse = !['failure', 'router', 'success', 'listen'].includes(nodeType)
+  const hasContextMenu = !['failure', 'success'].includes(nodeType)
+
+  const debugInfo = getDebugInfo(node.name)
+
+  const renderContents = () => {
+    switch (nodeType) {
+      case 'action':
+        return <ActionContents node={node} editNodeItem={editNodeItem} />
```

Comment:
Do you merge Action and Execute in a subsequent PR ?

Reply:
They are not the same thing for now so it's still separate (local actions vs remote actions on the action server). Not sure how we should handle them

Reply:
We should take a decision asap, it's so confusing. 
I suggest we add that in telemetry and remove remote actions if no one uses them . Util we implement that properly. It's a totally half baked feature